src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -446,7 +446,10 @@ private void initDB(String path, String logPath) {
             fillmodelColumnHandles();
 
         } catch (Exception e) {
-            IOUtils.closeQuietly(db::close);
+            log.error("Error while initializing RocksDb", e);
+            if (db != null) {
+                IOUtils.closeQuietly(db::close);
+            }
```

Comment:
@GalRogozinski FYI - You don't need the null check here. IOUtils already checks for null and doesn't close if null:

input - the Reader to close, may be null or already closed

Reply:
It confused me as well. Note that since `db` doesn't implement `Closeable` you are passing a reference to the `close()` method from a presumably non-null instance.

 Here is a stack trace for your amusement:
```
c.i.i.s.r.RocksDBPersistenceProvider - Initializing Database Backend...
Apr 30 19:15:11 iota-pn-1 java[116121]: 04/30 19:15:11.548 [main] ERROR com.iota.iri.IRI - Exception during IOTA node initialisation:
Apr 30 19:15:11 iota-pn-1 java[116121]: java.lang.NullPointerException: null
Apr 30 19:15:11 iota-pn-1 java[116121]:         at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.initDB(RocksDBPersistenceProvider.java:458) ~[iri-1.4.2.4_RC.jar:na]
Apr 30 19:15:11 iota-pn-1 java[116121]:         at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.init(RocksDBPersistenceProvider.java:76) ~[iri-1.4.2.4_RC.jar:na]
Apr 30 19:15:11 iota-pn-1 java[116121]:         at com.iota.iri.storage.Tangle.init(Tangle.java:26) ~[iri-1.4.2.4_RC.jar:na]
Apr 30 19:15:11 iota-pn-1 java[116121]:         at com.iota.iri.Iota.init(Iota.java:105) ~[iri-1.4.2.4_RC.jar:na]
Apr 30 19:15:11 iota-pn-1 java[116121]:         at com.iota.iri.IRI.main(IRI.java:71) ~[iri-1.4.2.4_RC.jar:na]
```

Reply:
OMG 😲  I take that back! You are correct.

Reply:
I took your advice.
Created an `IotaIOUtils`

Reply:
I suppose, it probably should read `if(it!=null){ it.close();}` . And maybe also add a Logger to the class and log the exception if one is produced in close();

Reply:
good idea